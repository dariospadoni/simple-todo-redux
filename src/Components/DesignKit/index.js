import styled from 'styled-components';

const HEADER_BG = '#eceff1';
const BORDER_COLOR = '#ddd';
const BUBBLE_BG = '#00B0FF';

const ChatApp = styled.div`
  border-radius: 5px;
  border: 2px solid ${BORDER_COLOR};
  color: #1A1A1A;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 97vh;
  margin: 10px;
  min-width: 320px;
  font-family: "Times New Roman", Times, serif;
`;
ChatApp.displayName = 'ChatApp';

const ChatHeader = styled.div`
  background-color: ${HEADER_BG};
  border-bottom: 2px solid ${BORDER_COLOR};
  display: flex;
  justify-content: space-between;
  padding: 10px;
  
  h1 {
    font-size: 20px;
    margin: 0;
    flex: 1;
  }
  
  .counter {
    margin-left: auto;
  }
`;
ChatHeader.displayName = 'ChatHeader';

const ChatMessages = styled.div`
  flex: 1;
  margin: 10px;
  overflow: scroll;
`;
ChatMessages.displayName = 'ChatMessages';

const ChatBubble = styled.div`
  background: ${BUBBLE_BG};
  border-radius: 7px;
  color: white;
  display: flex;
  margin-bottom: 10px;
  margin-left: 10px;
  position: relative;

  &:before {
    border: 10px solid transparent;
    border-right-color: ${BUBBLE_BG};
    border-left: 0;
    border-bottom: 0;
    content: '';
    height: 0;
    left: 0;
    margin-top: -10px;
    margin-left: -10px;
    position: absolute;
    top: 50%;
    width: 0;
  }
`;
ChatBubble.displayName = 'ChatBubble';

const ChatEditForm = styled.form`
  padding: 10px;
  width: 100% ;
  
  input {
    border: none;
    border-radius: 5px;
    line-height: 30px;
    width: 100%;
    
    &:focus {
      outline: none;
    }
  }
`;

const ChatMessage = styled.div`
  flex-grow: 1;
  padding: 10px;
`;

const ChatText = styled.div``;

const ChatTimeStamp = styled.small`
  color: #efefef;
  display: block;
  font-size: 12px;
  margin-left: 0px;
  margin-top: 10px;  
`;

const ChatAction = styled.a`
  cursor: pointer;
  flex: 0;
  font-size: 12px;
  padding: 24px 6px;
 
  &:hover {
    color: yellow;
    font-size: 13px;
  }
`;

const ChatInputForm = styled.form`
  border-top: 2px solid ${BORDER_COLOR};
  padding: 10px 5px;
  
  input {
    background-color: ${HEADER_BG};
    border: 1px solid ${BORDER_COLOR};
    border-radius: 5px;
    box-sizing: border-box;
    padding: 10px;
    width: 100%;
  }
`;

export {
  ChatInputForm, ChatText, ChatMessage, ChatTimeStamp, ChatAction, ChatEditForm, ChatBubble, ChatApp, ChatHeader, ChatMessages,
};
