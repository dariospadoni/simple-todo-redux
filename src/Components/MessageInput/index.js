import React from 'react';
import PropTypes from 'prop-types';
import { ChatInput } from '../DesignKit';

class MessageInput extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = { value: '' };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(evnt) {
    this.setState({ value: evnt.target.value });
  }

  onSubmit(evnt) {
    evnt.preventDefault();
    this.props.onSubmit(this.state.value);
    this.setState({ value: '' }); // reset input on submit
  }

  render() {
    return (
      <ChatInput onSubmit={this.onSubmit}>
        <input onChange={this.onChange} value={this.state.value} type="text" placeholder="Insert a todo..." />
      </ChatInput>
    );
  }
}


MessageInput.propTypes = {
  onSubmit: PropTypes.func,
};

export default MessageInput;
