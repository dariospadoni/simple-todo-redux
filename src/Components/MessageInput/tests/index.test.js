import React from 'react';
import { render, shallow } from 'enzyme';
import MessageInput from '../index';
import { ChatInput } from '../../DesignKit/index';


describe('MessageInput component', () => {
  it('should render the component', () => {
    const mockFn = jest.fn();
    const component = shallow(<MessageInput value="test" onSubmit={mockFn} />);
    expect(component.find(ChatInput).exists()).toBe(true);
  });

  it('should submit on enter', () => {
    const mockFn = jest.fn();
    const component = shallow(<MessageInput value="test" onSubmit={mockFn} />);
    component.find(ChatInput).simulate('submit', {
      preventDefault: () => {},
    });
    expect(mockFn).toHaveBeenCalled();
  });

  it('render snapshot', () => {
    const mockFn = jest.fn();
    const component = render(<MessageInput value="test" onSubmit={mockFn} />);
    expect(component).toMatchSnapshot();
  });
});
