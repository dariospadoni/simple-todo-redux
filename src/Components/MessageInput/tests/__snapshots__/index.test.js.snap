// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageInput component render snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "DesignKit__ChatInput-cnWgab huzDUa",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "placeholder": "Insert a todo...",
          "type": "text",
          "value": "",
        },
        "children": Array [],
        "name": "input",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "placeholder": undefined,
          "type": undefined,
          "value": undefined,
        },
        "x-attribsPrefix": Object {
          "placeholder": undefined,
          "type": undefined,
          "value": undefined,
        },
      },
    ],
    "name": "form",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
